### ISSUE TITLE
False Positive: IL_INFINITE_LOOP incorrectly reported for terminating loops

### ISSUE THREAD

**Issue by leemeii** [spotbugs/spotbugs#3904]
**Describe the bug**
The rule **IL_INFINITE_LOOP** incorrectly flags terminating inner loops as infinite when they are nested within a true infinite outer loop. While the outer loop is indeed infinite, the inner loop has a clear termination path and should not be flagged.

**To Reproduce**

```java
package test;

public class Bug {

    public void test() {
        int i = 1;
            do {
                int j = 1;
                do {
                } while (j < 1);//<- Should Not Reporte (False Positive)
     
            } while (i < 5); //<- reported correctly
        
    }
}
```

**Expected behavior**
SpotBugs should report IL_INFINITE_LOOP  **only** for the outer while (i < 5) loop. It should **NOT** report it for the inner loop (while (j < 1)), as the inner loop logic is self-terminating regardless of the outer loop's state.

**Actual behavior**
Both of the outer loop and the inner loop are reported as IL_INFINITE_LOOP.

**Environment**
- Java version:JDK 17
- SpotBugs Version:4.9.8
- How do you use SpotBugs: CLI
